gluten-data/src/main/java/io/glutenproject/vectorized/VeloxColumnarBatchBuilderJniWrapper.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.vectorized;
+
+import io.glutenproject.init.JniInitialized;
+
+public class VeloxColumnarBatchBuilderJniWrapper extends JniInitialized {
+  public static VeloxColumnarBatchBuilderJniWrapper INSTANCE =
+      new VeloxColumnarBatchBuilderJniWrapper();
+
+  public native long createVeloxColumnarBatch(long memoryManagerHandle, long cSchema, long cArray);
+}
```

Comment:
We had an implicit convention to avoid Velox-related code from appearing in `gluten/data`

There was an issue related https://github.com/oap-project/gluten/issues/1539

Regarding this code, IIUC we do `byte-array -> arrow -> velox` conversion, is it possible to just pass a native Arrow batch (created using `ColumnarBatchJniWrapper.INSTANCE.createWithArrowArray`) to native, and let Velox backend do lazy loading (which means the loading procedure from native Arrow batch to native Velox batch)?

Reply:
>  is it possible to just pass a native Arrow batch (created using ColumnarBatchJniWrapper.INSTANCE.createWithArrowArray) to native, and let Velox backend do lazy loading

The first version, that's how I did it. However, I can not find a good way to let downstream operator to do the lazy conversion(from arrow to velox ColumnarBatch). So, I create a new Jni method and convert it to velox ColumnarBatch directly.

Do you have any idea about it?

Reply:
Ideally it should be done implicitly in C++ Velox backend by using existing API `VeloxColumnarBatch::from`. If it's not working we may need to change the C++ backend codes to support that somehow, probably in another PR separately.